src/App.tsx

```diff
@@ -1,14 +1,78 @@
-/* eslint-disable max-len */
-import React from 'react';
+/* eslint-disable no-extra-boolean-cast */
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+import { getTodos, getUser } from './api';
 
 export const App: React.FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [filteredBy, setFilteredBy] = useState('all');
+  const [searchValue, setSearchValue] = useState('');
+  const [isDeleteButtonAval, setDeleteButtonAval] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [isModalShown, setModalShown] = useState(false);
+  const [isDataLoaded, setDataLoaded] = useState('isLoading');
+  const [isTodoLoaded, setTodoLoaded] = useState('isLoading');
```

Comment:
For loading is better use boolean value.
For Error beter have separate state.
isTodoLoaded - is useless, you can use `todo` state instead. If  `!todo && !error` - show loader. Something like this